shell/platform/common/cpp/engine_switches.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/common/cpp/engine_switches.h"
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+
+namespace flutter {
+
+std::vector<std::string> GetSwitchesFromEnvironment() {
+  std::vector<std::string> switches;
+  // Read engine switches from the environment in debug/profile. If release mode
+  // support is needed in the future, it should likely use a whitelist.
+#ifndef FLUTTER_RELEASE
```

Comment:
Is there somewhere that we're documenting this for users?

I'm trying to figure out how a user will know that this works in debug/profile but not in release (and how a user will know that they have to prefix the settings with `FLUTTER_ENGINE_SWITCH_`). 

I'm fine with landing this patch without that documentation, but it seems like something that we should try to address somehow. I have no idea where the right place for that would be. I'm somewhat tempted to say we should log a message if we encounter a swtich in debug/profile mode to let the user know that this setting will not be applied in release mode - but I have to admit my own use cases for these would be things like turning off the observatory service auth code or fixing the port to a set port each run, which would never apply in release mode anyway.

Reply:
I was trying to figure out if there's anywhere we document the current iOS and Android solutions (command line arguments and intents); I didn't see anything. If there is, then once desktop documentation is no longer confined to flutter.dev/desktop we would add the desktop documentation there.

The main way I expect this to benefit users, and for them to get benefits from this that have documentation, is fixing https://github.com/flutter/flutter/issues/66532.

Reply:
I don't think we have any docs for Android/iOS - in fact, I think most of this is handled by the tool by setting various intent flag arguments or command line arguments (android/iOS respectively), in which case it's probably fine to just leave as is, since presumably the tool will know what it's doing here.

Reply:
> I'm trying to figure out how a user will know that this works in debug/profile but not in release

An open question is whether this is a use case that will ever actually come up. If so, I suspect we'll need whitelisted support for release, per the comment in the code, rather than just documentation. We'll find out if/when someone files a bug :)

(The fact that it took so long after desktop existed for https://github.com/flutter/flutter/issues/66532 and https://github.com/flutter/flutter/issues/60393 to be filed suggests to me that these use cases are quite rare, so seeing what's actually requested seems like the best way to figure out what to do with release mode.)

Reply:
Yeah, these will primarily be used by the tool, which is fine.